src/Ledger/Conway/Specification/PParams.lagda.md

```diff
@@ -32,102 +36,177 @@ module Ledger.Conway.Specification.PParams
+   settings;
+
++  `GovernanceGroup`{.AgdaInductiveConstructor}: parameters related to governance
+   settings;
+
++  `SecurityGroup`{.AgdaInductiveConstructor}: parameters that can impact the
+   security of the system.
 
-\begin{NoConway}
-The \AgdaRecord{Acnt} record has two fields, \AgdaField{treasury} and \AgdaField{reserves}, so
-the \AgdaBound{acnt} field in \AgdaRecord{NewEpochState} keeps track of the total assets that
-remain in treasury and reserves.
+The purpose of these groups is to determine voting thresholds for proposals aiming to
+change parameters.  Given a proposal to change a certain set of parameters, we look
+at which groups those parameters fall into and from this we determine the voting
+threshold for that proposal.  (The voting threshold calculation is described in
+detail in the [Ratification Requirements][] section.)
 
-\begin{figure*}[ht]
-\begin{AgdaMultiCode}
-\begin{code}
+The first four groups have the property that every protocol parameter is associated
+to precisely one of these groups.  The `SecurityGroup`{.AgdaInductiveConstructor} is
+special: a protocol parameter may or may not be in the
+`SecurityGroup`{.AgdaInductiveConstructor}.  So, each protocol parameter belongs to
+at least one and at most two groups.  Note that in
+[CIP-1694](https://cips.cardano.org/cip-1694) there is no
+`SecurityGroup`{.AgdaInductiveConstructor}, but there is the concept of
+security-relevant protocol parameters (see [CKB+23](#cip1694)).  The difference
+between these notions is only social, so we implement security-relevant protocol
+parameters as a group.
+
+```agda
 record Acnt : Type where
-\end{code}
-\begin{code}[hide]
+```
+
+<!--
+```agda
   constructor ⟦_,_⟧ᵃ
-\end{code}
-\begin{code}
+```
+-->
+
+```agda
   field
     treasury reserves : Coin
+```
 
+<!--
+```agda
 instance
   HasTreasury-Acnt : HasTreasury Acnt
   HasTreasury-Acnt .TreasuryOf = Acnt.treasury
 
   HasReserves-Acnt : HasReserves Acnt
   HasReserves-Acnt .ReservesOf = Acnt.reserves
+```
+-->
 
+```agda
 ProtVer : Type
 ProtVer = ℕ × ℕ
+```
 
+<!--
+```agda
 instance
   Show-ProtVer : Show ProtVer
   Show-ProtVer = Show-×
+```
+-->
 
+```agda
 data pvCanFollow : ProtVer → ProtVer → Type where
   canFollowMajor : pvCanFollow (m , n) (m + 1 , 0)
   canFollowMinor : pvCanFollow (m , n) (m , n + 1)
-\end{code}
-\end{AgdaMultiCode}
-\caption{Definitions related to protocol parameters}
-\label{fig:protocol-parameter-defs}
-\end{figure*}
-\end{NoConway}
-\begin{code}[hide]
+```
+
+<!--
+```agda
 instance
   unquoteDecl HasCast-Acnt = derive-HasCast
     [ (quote Acnt , HasCast-Acnt) ]
-\end{code}
+```
+-->
 
-\begin{figure*}[ht]
-\begin{code}
+
+### Protocol Parameter Group Definition {#sec:protocol-parameter-group-definition}
+
+```agda
 data PParamGroup : Type where
   NetworkGroup     : PParamGroup
   EconomicGroup    : PParamGroup
   TechnicalGroup   : PParamGroup
   GovernanceGroup  : PParamGroup
   SecurityGroup    : PParamGroup
```

Comment:
This is far away from whee the groups are described

Reply:
It seems far, but because of all the hidden blocks, it's actually not too far down on the page from the description.  I think it's okay.